src/components/Journey/PageModal.tsx

```diff
@@ -20,69 +24,117 @@ const style = {
+
+//to pass to category page for checking status
+export const sub = signal<SignalValue>({ name: "", status: "" });
+export const pantrySignal = signal<PantrySignalValue>({ document: [] });
+
 export default function PageModal({
   product,
   subCategory,
 }: {
-  product: ProductDetails;
-  subCategory: { key: string; product: ProductDetails[]; status: string };
+  product: Models.Document;
+  subCategory: PantryProductDocument | undefined;
 }) {
-  const { setPantryProducts } = useContext(PantryContext);
+  //to save into db
+  // const [pantryData, setPantryData] = useState<string[]>([]);
+  const [, setPantryData] = useState<string[]>([]);
+  const [userDoc, setUserDoc] = React.useState("");
+
   const { setCategoryStatus } = useContext(CategoryStatusContext);
+
   const [open, setOpen] = React.useState(false);
   const handleOpen = () => setOpen(true);
   const handleClose = () => setOpen(false);
+
+  const checkCompleteStatus = () => {
+    return subCategory?.value.every(
+      (product: Models.Document) => product.Status === STATUS.ACTIVE,
+    );
+  };
+
+  const checkActiveStatus = () => {
+    return subCategory?.value.some(
+      (product: Models.Document) => product.Status === STATUS.ACTIVE,
+    );
+  };
+
   const addToJourney = () => {
     //update current product's status to active
-    product.status = STATUS.ACTIVE;
+    if (subCategory) {
+      product.Status = STATUS.ACTIVE;
+      // setPantryData((prevItem) => [...prevItem, product.$id]);
+      setPantryData((prevItem) => [...prevItem, product.$id]);
+      pantrySignal.value = {
+        document: pantrySignal.value.document
+          ? [...pantrySignal.value.document, product]
+          : [product],
+      };
 
-    setPantryProducts([
-      {
-        key: subCategory,
-        value: product,
-      },
-    ]);
+      //update category's status depends on current subCategory's product list length and each status
+      if (checkCompleteStatus()) {
+        subCategory.status = STATUS.COMPLETED;
+        sub.value = { name: subCategory.key, status: STATUS.COMPLETED };
+        // sub.value = subCategory.status; //to delete from improve products section
+        setCategoryStatus({
+          category: subCategory.key,
+          status: `${STATUS.COMPLETED} (${subCategory.key})`,
+        });
+      } else if (checkActiveStatus()) {
+        subCategory.status = STATUS.ACTIVE;
+        setCategoryStatus({
+          category: subCategory.key,
+          status: `${STATUS.ACTIVE} (${subCategory.key})`,
+        });
+      } else {
+        subCategory.status = STATUS.NONE;
+        setCategoryStatus({
+          category: subCategory.key,
+          status: `${subCategory.key} not started`,
+        });
+      }
 
-    const checkCompleteStatus = () => {
-      return subCategory.product.every(
-        (product) => product.status === STATUS.ACTIVE,
-      );
-    };
+      //get current user and update user's pantry data
+      const fetchData = async () => {
```

Comment:
Instead of this, you can use the `useUserData` hook. This has a cached profile of the user you can use.

Reply:
thanks for pointing it out. I used useUserData in Profile and forgot to replace it in the PageModal page.